frontend/src/pages/users/ViewProfile.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { Box, Text, useColorModeValue, HStack, IconButton, Flex } from '@chakra-ui/react';
+import { EmailIcon, EditIcon } from '@chakra-ui/icons';
+import { FaUserGroup, FaCode } from 'react-icons/fa6';
+import { useAppSelector } from '../../reducers/hooks';
+import { selectUser } from '../../reducers/authSlice';
+import EditProfile from './EditProfile';
+import DeregisterButton from '../auth/DeregisterButton';
+
+const ViewProfile: React.FC = () => {
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+
+  const user = useAppSelector(selectUser);
+
+  return (
+    <Box width="100%" height="85vh" my={5}>
+      <Box
+        bg={useColorModeValue('gray.50', 'gray.700')}
+        boxShadow="lg"
+        borderWidth="2px"
+        borderRadius="lg"
+        p={4}
+        width={['90%', '80%', '70%', '50%']}
+        mx="auto"
+        position="relative"
+      >
+        <Text fontSize="xl" fontWeight="bold" textAlign="center">
+          {user?.username}
+        </Text>
+        <IconButton
+          isRound={true}
+          variant="solid"
+          colorScheme="gray"
+          aria-label="Edit Profile"
+          icon={<EditIcon />}
+          position="absolute"
+          top="4"
+          right="4"
+          zIndex="1"
+          onClick={() => {
+            setIsEditModalOpen(true);
+          }}
+        />
+        <EditProfile
+          isOpen={isEditModalOpen}
+          onCloseModal={() => {
+            setIsEditModalOpen(false);
+          }}
+          initialUsername={user?.username !== undefined ? user?.username : ''}
+          initialEmail={user?.email !== undefined ? user?.email : ''}
+          initialLanguages={user?.languages !== undefined ? user?.languages : []}
+        />
+        <Box textAlign="left" pl={12} pr={12} pt={2} pb={8}>
+          <HStack spacing={5} align="center" mt={4}>
+            <EmailIcon boxSize={6} />
+            <Text>
+              <span style={{ fontWeight: 'bold' }}>Email: </span>
+              {user?.email}
+            </Text>
+          </HStack>
+          <HStack spacing={5} align="center" mt={4}>
+            <FaUserGroup fontSize="24px" />
+            <Text>
+              <span style={{ fontWeight: 'bold' }}>Role: </span>
+              {user?.role}
+            </Text>
+          </HStack>
+          <HStack spacing={5} align="center" mt={4}>
+            <FaCode fontSize="24px" />
+            <Text>
+              <span style={{ fontWeight: 'bold' }}>Languages: </span>
+              {(user?.languages?.length ?? 0) > 0 ? user?.languages?.map((lang) => lang.language).join(', ') : 'None'}
+            </Text>
+          </HStack>
```

Comment:
nit: icon seems to shrink if there are many languages
<img width="778" alt="image" src="https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g17/assets/34371483/348468fe-6928-4d2a-af90-d970b4dd55c5">